src/rest_framework_reactive/consumers.py

```diff
@@ -69,23 +127,44 @@ def orm_notify_table(self, message):
             .values_list('pk', flat=True)
         )
 
-        for observer in observers:
+        for observer_id in observers:
             async_to_sync(self.channel_layer.send)(
                 CHANNEL_WORKER_NOTIFY,
-                {'type': TYPE_EVALUATE_OBSERVER, 'observer': observer},
+                {'type': TYPE_EVALUATE_OBSERVER, 'observer': observer_id},
             )
 
     def observer_evaluate(self, message):
         """Evaluate observer."""
+        observer_id = message['observer']
+        throttle_rate = get_queryobserver_settings()['throttle_rate']
+
         try:
-            observer = Observer.objects.only('pk', 'request').get(
-                pk=message['observer']
-            )
+            observer_qs = Observer.objects.only('pk', 'request')
+            observer = observer_qs.get(pk=observer_id)
         except Observer.DoesNotExist:
             return
 
+        if throttle_rate <= 0:
+            pass
+        elif is_throttled(observer_id):
+            """Discard the evaluation since another is waiting."""
+            return
+        elif is_evaluated(observer_id):
+            """Schedule observer to be throttled."""
+            # Flag (in cache) that observer is being throttled
+            observer_throttled_set(observer_id)
+            async_to_sync(self.channel_layer.send)(
+                CHANNEL_THROTTLE, {'type': TYPE_THROTTLE, 'observer': observer_id}
+            )
```

Comment:
A `return` is missing here? Also add a test for this.

Reply:
It would be nicer if you put all logic inside the if statements - this means a duplicated `_evaluate` line, but use-cases are too distinct for combining them in a reasonable way.